Core/ELF/ElfReader.cpp

```diff
@@ -29,8 +29,8 @@ const char *ElfReader::GetSectionName(int section) const {
 		return nullptr;
 
 	int nameOffset = sections[section].sh_name;
-	if (nameOffset < 0) {  // TODO: Where can we get a solid upper limit?
-		ERROR_LOG(LOADER, "ELF: Bad name offset %d in section %d", nameOffset, section);
+	if (nameOffset < 0 || nameOffset >= size_) {
```

Comment:
Sorry, I mean that there's no guarantee a null terminated string exists here.  Let's say that nameOffset is 3:

```
A B C D E F G H I J K L M N O P Q R S T U X X X X X
    ^                         ^
    \-- Start of string       \--- End of ELF
```

No one promised there'd be a NUL byte here anywhere.  We might still, when strcmp()ing with the string, go past size_.

Obviously, if the first byte of `nameOffset` is 0, that'd be fairly stupid.

-[Unknown]